### PR TITLE
Association

### PR DESCRIPTION
**Pull request by MICHAELMUNAVU83**
- Run a program using the command line.
- Set up associations between classes and objects.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
 class Student < Person
-  attr_accessor :classroom
+  attr_reader :classroom, :parent_permission, :name, :age
 
-  def initialize(classroom, age, name = 'Unknown', parent_permission: true)
-    super(age, name, parent_permission)
+  def initialize(age, name, classroom, parent_permission: true)
+    super(age, name, parent_permission: parent_permission)
     @classroom = classroom
   end
```
Comment (medaminedev66) [inline student.rb:10]: - [ ]  In the Student class, there should be a method that serves as the 'belongs-to' side (a student belongs to a classroom). Check out [this article with examples](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) for more information.

File: person.rb [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
+  attr_accessor :name, :age, :rentals
 
   def initialize(age, name = 'Unknown', parent_permission: true)
-    super()
-    @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    super()
   end
```
Comment (medaminedev66) [inline person.rb]: - [ ] In the Person class, `@rentals` should be initialized according to the `a person has many rentals` relationship. Check out [the example](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars) about constructors in a previous lesson.

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rentals(date, person, _book)
+    Rental.new(date, person, self)
+  end
```
Comment (medaminedev66) [inline book.rb]: - [ ] There is no need to the parameter `_book` since it's not used in the method.

File: person.rb [hunk 4]
```diff
@@ -1,28 +1,35 @@
...
     of_age? || @parent_permission
   end
+
+  def correct_name
+    @name
+  end
+
+  def add_rentals(date, _person, book)
+    Rental.new(date, self, book)
+  end
```
Comment (medaminedev66) [inline person.rb]: - [ ] There is no need for the parameter `_book` since it's not used in the method.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @MICHAELMUNAVU83   😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- The repo looks professional
- There are no linter errors
- Your code is clean.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @MICHAELMUNAVU83, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**